### ISSUE TITLE
A pop up saying something about my data being synced pops up and interrupts my typing

### ISSUE THREAD

**Issue by Enderform3** [stravo1/obsidian-gdrive-sync#45]
the title pretty much sums it up.

**Comment by stravo1** [comment]
if you're talking about the "... has been externally modified, merging changes automatically" popup then unfortunately can't do much about it. you can type faster or increase the refresh time to reduce it's occurrence but that's it. although it's a cosmetic issue, should not affect typing or syncing. 

what do you mean by interrupting? it shouldn't really interrupt your typing.

**Comment by somogyi-ede** [comment]
I'm having the same issue and I can confirm that it takes focus away from what you're typing whenever the message pops up.
I installed your plugin on Windows, macOS and Android and the issue occurs even if I don't have any other instances of Obsidian open on any of my other devices. Unfortunately it's not a solely cosmetic issue.

**Comment by stravo1** [comment]
yes, i have noticed it on tables, even sometimes for Latex expressions it takes away focus. unfortunately i don't know how to solve it. but for normal texts it's fine.

**Comment by stravo1** [comment]
https://github.com/user-attachments/assets/61ef7bd1-99b1-4901-983f-dcdb74539030

**Comment by somogyi-ede** [comment]
Yes, I've been working with tables lately and I noticed this behavior when writing to a table -- regular typing is indeed not affected, as your video suggests. Again, it may not even be from the popup as I've been seing it less since redownloading all my files and removing all Obsidian config files inside `.obsidian`.

The focus loss inside tables definitely has something to do with the syncing interval though. I set it to a minute and the issue was gone, but only for a minute.

There seem to be two separate, but related issues here. I did some more testing and found that both the popup and the syncing take the focus away:
- the syncing by itself after a few seconds with the default 5-second sync interval set, but not while typing
- and also the popup, which can even take focus away _while_ typing.

**Comment by stravo1** [comment]
the pop up comes while the sync tries to refresh the content while you are typing, so the losing of focus during pop up is a subset of the main issue where the lose of focus happens during the sync as the pop up comes during sync only.

**Comment by stravo1** [comment]
most probably it is out of my control but as you suggested increasing the refresh time should help

**Comment by somogyi-ede** [comment]
Unfortunately this is going to be a dealbreaker for a lot of people, which is a shame, as it's a useful little plugin.
Increasing the sync interval is not really a solution as it's rather inconvenient (you may lose work if you set it too high and quit Obsidian without waiting out those X seconds) and it doesn't help with the popup taking focus away (as opposed to the syncing mechanism where there's no popup, see above).
Please, if you can, take a look and see if there's anything that can be done on your end.
Do we know if the same thing occurs with Obsidian's own sync?

**Comment by stravo1** [comment]
will try my best to figure something out, thanks for the detailed reporting.

updates are announced in the discord server. also i need testers for testing new features. so if you don't mind please do join the discord server.

**Comment by somogyi-ede** [comment]
Sure, let me join. Thanks!

**Comment by stravo1** [comment]
https://discord.com/invite/dPasX4Ac2P

**Comment by Enderform3** [comment]
> will try my best to figure something out, thanks for the detailed reporting.
> 
> updates are announced in the discord server. also i need testers for testing new features. so if you don't mind please do join the discord server.

thanks for responding! Maybe use some kind of log in the properties to see if the file was written from the current device and not update the information on the page currently being written. This would either solve the problem, or make it slightly less annoying.

**Comment by jennos** [comment]
Has anything been done to address this issue? I often find myself using tables and this bug makes it basically impossible to use tables effectively.

**Comment by stravo1** [comment]
@jennos @Enderform3 @somogyi-ede a fix has been released for this issue on the beta-19 release, please do update the plugin and let me know how it goes.

**Comment by tholness** [comment]
Great tool. Wondering if is there any progress on resolving this fully? I use tables a lot and on a desktop Issue 75 also becomes a pain to deal with. Thanks
